### PR TITLE
Bug 1368883 added info on resource limits in downward API

### PR DESCRIPTION
**Pull request by bfallonf**
As per: https://bugzilla.redhat.com/show_bug.cgi?id=1368883

@aveshagarwal FYI

I'll comment in the BZ.

### CODE REVIEW

**Comment by bfallonf** [comment]
@aveshagarwal FYI

File: dev_guide/downward_api.adoc [hunk 1]
```diff
@@ -222,3 +219,139 @@ annotation2=456
...
+          - path: "mem_request"
+            resourceFieldRef:
+              containerName: client-container
+              resource: requests.memory
+....
+----
+====
++
+If the resource limits are not included in the container configuration, the downward API defaults to the node's CPU and memory capacities.
+
```
Comment (aveshagarwal) [inline dev_guide/downward_api.adoc]: node's CPU and memory allocatables

It was changed in https://github.com/kubernetes/kubernetes/pull/29639 and I also sent a doc update PR here:
https://github.com/kubernetes/kubernetes.github.io/pull/1107/files

File: dev_guide/downward_api.adoc [hunk 2]
```diff
@@ -222,3 +219,139 @@ annotation2=456
...
+              resource: requests.memory
+        - name: MY_MEM_LIMIT
+          valueFrom:
+            resourceFieldRef:
+              resource: limits.memory
+...
+----
+====
++
+If the resource limits are not included in the container configuration, the downward API defaults to the node's CPU and memory capacities.
```
Comment (aveshagarwal) [inline dev_guide/downward_api.adoc]: node's CPU and memory allocatables

It was changed in kubernetes/kubernetes#29639 and I also sent a doc update PR here:
https://github.com/kubernetes/kubernetes.github.io/pull/1107/files

File: dev_guide/downward_api.adoc [hunk 3]
```diff
@@ -222,3 +219,139 @@ annotation2=456
...
+. When creating a pod configuration, use the `spec.volumes.downwardAPI.items` field to describe the desired resources that correspond to the `spec.resources` field:
++
+====
+----
+....
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox:1.24
+      command: [ "/bin/sh", "-c", "env" ]
```
Comment (aveshagarwal) [inline dev_guide/downward_api.adoc]: This example is not correct for volume plugin. 
see the file docs/user-guide/downward-api/volume/dapi-volume-resources.yaml
in  https://github.com/kubernetes/kubernetes.github.io/pull/591.

File: dev_guide/downward_api.adoc (hunk 1 above)
Comment (aveshagarwal) [inline dev_guide/downward_api.adoc]: node's CPU and memory allocatables

It was changed in kubernetes/kubernetes#29639 and I also sent a doc update PR here:
https://github.com/kubernetes/kubernetes.github.io/pull/1107/files

File: dev_guide/downward_api.adoc (hunk 2 above)
Comment (bfallonf) [inline dev_guide/downward_api.adoc]: @aveshagarwal I had to Google it, but it turns out allocatables is not a word... So I changed to "allocatable values".

File: dev_guide/downward_api.adoc (hunk 3 above)
Comment (bfallonf) [inline dev_guide/downward_api.adoc]: @aveshagarwal Right. Sorry, i took in the last one in order to save space. I've since changed back though.

**Comment by bfallonf** [comment]
@aveshagarwal I've made the changes as per your suggestions. I'll presume everything else is OK and move forward, but please let me know if there's any other issues with this. Thanks!

@adellape @ahardin-rh @tpoitras Probably OK for peer review please!

**Comment by bfallonf** [comment]
[rev_history]
|xref:../dev_guide/downward_api.adoc#dev-guide-downward-api[Downward API]
|Added information on consuming container resources using the downward API.
%

File: dev_guide/downward_api.adoc (hunk 2 above)
Comment (aveshagarwal) [inline dev_guide/downward_api.adoc]: Fine with me.

File: dev_guide/downward_api.adoc [hunk 4]
```diff
@@ -222,3 +219,146 @@ annotation2=456
...
++
+====
+----
+$ oc create -f volume-pod.json
+----
+====
+
+. To verify that the pod has been created with the expected values, you can grep
+the pod's logs and look for the following variables:
++
```
Comment (aveshagarwal) [inline dev_guide/downward_api.adoc]: This is not correct for the volume example above.

File: dev_guide/downward_api.adoc [hunk 5]
```diff
@@ -222,3 +219,146 @@ annotation2=456
...
+- `*MY_MEM_LIMIT*`
+- `*MY_CPU_LIMIT*`
+- `*MY_MEM_REQUEST*`
+- `*MY_CPU_REQUEST*`
+
+[[dapi-resources-volume-plugin]]
+=== Using the Volume Plug-in
+
+. When creating a pod configuration, use the `spec.volumes.downwardAPI.items`
```
Comment (ahardin-rh) [inline dev_guide/downward_api.adoc]: `*spec.volumes.downwardAPI.items*`

File: dev_guide/downward_api.adoc [hunk 6]
```diff
@@ -222,3 +219,146 @@ annotation2=456
...
+field to describe the desired resources that correspond to the `spec.resources`
```
Comment (ahardin-rh) [inline dev_guide/downward_api.adoc]: `*spec.resources*`

File: dev_guide/downward_api.adoc [hunk 7]
```diff
@@ -117,18 +115,17 @@ $ oc create -f pod.json
 ----
 ====
 
-. Check the container's logs:
+. Check the container's logs for the `MY_POD_NAME` and `MY_POD_NAMESPACE`
```
Comment (ahardin-rh) [inline dev_guide/downward_api.adoc]: `*MY_POD_NAME*`
    `*MY_POD_NAMESPACE*`

File: dev_guide/downward_api.adoc [hunk 8]
```diff
@@ -222,3 +219,146 @@ annotation2=456
...
+correctly create an image for specific environments.
+
+You can do this using both the xref:dapi-using-environment-variables[environment
+variable] and xref:dapi-using-volume-plugin[volume plug-in] methods.
+
+[[dapi-resources-environment-variables]]
+=== Using Environment Variables
+
+. When creating a pod configuration, specify environment variables that
+correspond to the contents of the `resources` field in the `spec.container`
```
Comment (ahardin-rh) [inline dev_guide/downward_api.adoc]: `*resources*`
    `*spec.container*`

**Comment by ahardin-rh** [comment]
@bfallonf just a few style nits from me. Other than that, looks good! :star2:

File: dev_guide/downward_api.adoc (hunk 4 above)
Comment (bfallonf) [inline dev_guide/downward_api.adoc]: @aveshagarwal OK. I removed this step.

**Comment by bfallonf** [comment]
Thanks @ahardin-rh suggestions put in. I'll merge away.

**Pull request merged**